CODE_OF_CONDUCT.adoc

```diff
@@ -0,0 +1,52 @@
+
+- https://raw.githubusercontent.com/eclipse/.github/master/CODE_OF_CONDUCT.md[Eclipse Code of Conduct]
+- https://www.eclipse.org/projects/handbook/[Eclipse Foundation Handbook]
+- https://gitlab.eclipse.org/eclipsefdn/emo-team/policies/legal-faqs/-/blob/main/documents/src/adoc/committers-dd.adoc[ Eclipse Committer Due Diligence Guidelines]
+
+Although many of the items discussed in this document is intuitive, we ask that you read and understand them anyways.
+
+
+We (the committers) aim to run this project in an open and fair manner where contributions are encouraged from all companies and individuals. If you happen to notice any violations to this code of conduct, please contact the project lead and/or the Eclipse Foundation Management Office (EMO).
+
+
+## Committers
+https://www.eclipse.org/projects/handbook/#roles-cm[Eclipse-uProtocol Committers] play a vital role to ensure ensure contributions from others (and themselves) follow the vision and mission of the project, and adhere to this code of conduct. 
+In the next section we will elaborate on their expected area of expertise and responsibilities, along with how committers are nominated and retired.
+
+[#area-of-expertise]
+### Committer's Area of Expertise
+Committers are chosen based on merit and the area of expertise that they can steward (maintain). List below is not exhaustive and shall update as the project evolves:
+
+1. **Specifications** (up-spec, up-core-api): Ensuring the vision/mission of the project is upheld, process is followed, and the specifications are maintained and updated. Concerns are discussed and addressed w.r.t backwards compatibility support and new features.
+2. **Programming language:** Experts in a particular language who help to review and comment on contributions for a given supported language (Java, Kotlin, Rust, C++, Python, etc...)
+3. **Implementation:** Experts in a particular implementation of uProtocol (ex. Android, Zenoh, Sommr/SOME/ip, Cloud, etc...)
+
+NOTE: Committers are strongly encouraged to contribute in more than one area defined above.
+
+
+### Responsibilities
+Additional roles and responsibilities of a uProtocol committers are as follows:
+
+* _Actively_ participate in weekly/bi-weekly project meetings
+* Makes _significant_ contributions in at least one of the <<area-of-expertise>> above
+* Contributes (code, reviews, discussions) to the specification and core API projects
+
+
+### Nomination
+Contributors are nominated by either the project lead or another committer per the https://www.eclipse.org/projects/handbook/#elections-committer[Eclipse Committer Nomination Process].
+
+
+### Retirement
+Per https://www.eclipse.org/projects/handbook/#elections-retire-cm[Eclipse Foundation Handbook], Committers may retire for one of the following reasons:
+
+1. Their own volition
+2. By the project lead (with supporting justification)
```

Comment:
we should probably define list or criteria for justification to be accepted. e.g. 
- active or passive obstruction to progress in discussions
- poor code quality
- intentional damage to the "assets" of project(s) under uProtocol
- inactive for long period of time (not mandatory to retire a committer for such reason, but a possibility)
- other?

Reply:
I re-worded a lot of the document, hopefully it is better now